### PR TITLE
Components widget basic architecture

### PR DESCRIPTION
**Pull request by Infinityworld13**
Added basic architecture outline for the Widget component

## Context

In order to rework the current Client User Interface and include the embed widget, we are making new react components and reworking the front end architecture.

## Your Changes

**Description**:

Added slideshow class and tradeInInput and carSlides react functions with their stares, props, and basic methods/constructors. Momentarily all of these are empty and do not generate anything when called or utilized in the UI.

**Type of change** (select all that apply):

- [x] New feature (additional functionality separate from existing code)
- [x] User Interface change (design changes)

## Testing

Don't require testing for now, as it's just the outline for future work.

## Checklist

- [x] Code is up to date with main branch (no merge conflicts)
- [ ] Code has been properly tested
- [ ] Code has been properly documented
- [ ] One or two people have reviewed this pull request

### CODE REVIEW

**Comment by leenlab2** [comment]
@Infinityworld13 I think you may have pushed your `.idea/` folder by accident, can you please remove it from this PR?

File: .idea/.gitignore [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Default ignored files
```
Comment (leenlab2) [inline .idea/.gitignore:1]: This file should not be included in the PR

File: src/components/widget/carSlides.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import React from "react";
+
+function CarSlides(){
+    const model=''
```
Comment (leenlab2) [inline src/components/widget/carSlides.js:4]: Now that I think about it, these might be better as props rather than states. Check the equivalent Listitemcar component in my PR.

Comment (leenlab2) [reply src/components/widget/carSlides.js:4]: You might want to read more about state hooks for function components: <https://reactjs.org/docs/hooks-state.html>

File: src/components/widget/slideshow.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
+import CarSlides from './carSlides'
+
+class Slideshow extends React.Component{
+    constructor(props) {
+        super(props);
+        this.state = {curr_slide: CarSlides};
+    }
```
Comment (leenlab2) [inline src/components/widget/slideshow.js:8]: Can you add a render method that calls on <CarSlides />, or at least a TODO if you cannot figure it out?

**Review by leenlab2 — COMMENTED** [review]
Good start, left some minor comments in addition to the point about removing the .idea folder

**Comment by Infinityworld13** [comment]
> @Infinityworld13 I think you may have pushed your `.idea/` folder by accident, can you please remove it from this PR?

Did it just now

**Review by dhruvaa02 — APPROVED** [review]
If everything Leen said was resolved then I see no issues.
